packages/select/src/components/omnibar/omnibar.tsx

```diff
@@ -65,7 +65,14 @@ export class Omnibar<T> extends React.PureComponent<IOmnibarProps<T>> {
 
     public render() {
         // omit props specific to this component, spread the rest.
-        const { initialContent = null, isOpen, inputProps, overlayProps, ...restProps } = this.props;
+        const {
+            initialContent = null,
+            isOpen,
+            inputProps,
+            overlayProps,
+            resetOnQuery = true,
```

Comment:
hmm actually this default does not belong here, it should be in `QueryList` itself so it applies to all components. you can revert this change as it was already passed through the spread.

Reply:
it's only used the [one time](https://github.com/palantir/blueprint/pull/2894/files/0cfe12eaf3249baa6b49b8eb1eec5f5f40d02eb6#diff-3ddddce91cf28ef273c52500667edb8eR185) in QueryList so you could just modify that method to do the deconstruct-default like you have here.

Reply:
Unless there's some es6 fanciness that I don't know. If I just deconstruct I have to do this:

      public setQuery(query: string, resetActiveItem?: boolean) {
        const { resetOnQuery } = this.props;

        // only override resetActiveItem if param is undefined
        if (typeof resetActiveItem === 'undefined') {
            // default to true if resetOnQuery prop is not defined
            if (typeof resetOnQuery === 'undefined') {
                resetActiveItem = true;
            }
            else {
                resetActiveItem = resetOnQuery;
            }
        }
        // ...rest of fn
      }

so I think I'll add a `public static defaultProps` to the component

    public static defaultProps = {
        resetOnQuery: true,
    };
    // ...later on...
    public setQuery(query: string, resetActiveItem = this.props.resetOnQuery) {